File: app/api/search_routes.py
from fastapi import APIRouter, HTTPException
from app.services.vector_store import search_emails

router = APIRouter()


@router.get("/search")
def search(q: str, top_k: int = 5):
    if not q.strip():
        raise HTTPException(status_code=400, detail="Query cannot be empty")

    results = search_emails(q.strip(), top_k)
    return {"query": q, "results": results}

File: backend/src/routes/search.routes.js
import { Router } from 'express';
import { searchEmails } from '../services/vector.service.js';
import { requireSession } from '../middleware/auth.middleware.js';

const router = Router();

router.get('/search', requireSession, async (req, res) => {
  const { q, top_k = 5 } = req.query;
  if (!q?.trim()) {
    return res.status(400).json({ detail: 'Query cannot be empty' });
  }

  try {
    const results = await searchEmails(req.session.email, q.trim(), Number(top_k));
    res.json({ query: q, results });
  } catch (err) {
    console.error('SEARCH ERROR:', err.message);
    res.status(500).json({ detail: err.message });
  }
});

export default router;
